backend/src/net/static_server.rs

```diff
@@ -274,69 +290,56 @@ async fn main_embassy_ui(req: Request<Body>, ctx: RpcContext) -> Result<Response
-                            None => Ok(bad_request()),
-                            _ => Ok(not_found()),
-                        }
-                    } else {
-                        Ok(not_found())
-                    }
+            if_authorized(&ctx, &request_parts, || async {
+                let sub_path = Path::new(path);
+                if let Ok(rest) = sub_path.strip_prefix("package-data") {
+                    FileData::from_path(
+                        &request_parts,
+                        &ctx.datadir.join(PKG_PUBLIC_DIR).join(rest),
+                    )
+                    .await?
+                    .into_response(&request_parts)
+                    .await
+                } else {
+                    Ok(not_found())
                 }
-                Err(e) => un_authorized(e, &format!("public/{path}")),
-            }
+            })
+            .await
         }
         (&Method::GET, Some(("proxy", target))) => {
-            match HasValidSession::from_request_parts(&request_parts, &ctx).await {
-                Ok(_) => {
-                    let target = urlencoding::decode(target)?;
-                    let res = ctx
-                        .client
-                        .get(target.as_ref())
-                        .headers(
-                            request_parts
-                                .headers
-                                .iter()
-                                .filter(|(h, _)| {
-                                    !PROXY_STRIP_HEADERS
-                                        .iter()
-                                        .any(|bad| h.as_str().eq_ignore_ascii_case(bad))
-                                })
-                                .map(|(h, v)| (h.clone(), v.clone()))
-                                .collect(),
-                        )
-                        .send()
-                        .await
-                        .with_kind(crate::ErrorKind::Network)?;
-                    let mut hres = Response::builder().status(res.status());
-                    for (h, v) in res.headers().clone() {
-                        if let Some(h) = h {
-                            hres = hres.header(h, v);
-                        }
+            if_authorized(&ctx, &request_parts, || async {
+                let target = urlencoding::decode(target)?;
+                let res = ctx
+                    .client
+                    .get(target.as_ref())
+                    .headers(
+                        request_parts
+                            .headers
+                            .iter()
+                            .filter(|(h, _)| {
+                                !PROXY_STRIP_HEADERS
+                                    .iter()
+                                    .any(|bad| h.as_str().eq_ignore_ascii_case(bad))
+                            })
+                            .map(|(h, v)| (h.clone(), v.clone()))
```

Comment:
❓  Isn't there an operator of cloned()
Might just be an Iterators crate?

Reply:
cloned is `impl Iterator<Item = &T> -> impl Iterator<Item = T>` not `impl Iterator<Item = (&T, &T)> -> impl Iterator<Item = (T, T)>`